### ISSUE TITLE
Circular recommendations: PLC0206, PERF102, SIM118

### ISSUE THREAD

**Issue by njsaunders** [astral-sh/ruff#14656]
I have the following code:

```
    for audit in perf_audits:
        max_audit_points = perf_audits[audit]
```

This raises PLC0206:

```
helpers/lighthouse.py:246:5: PLC0206 Extracting value from dictionary without calling `.items()`
    |
244 |       points_lost = {}
245 |
246 |       for audit in perf_audits:
    |  _____^
247 | |         max_audit_points = perf_audits[audit]
248 | |         acheived_audit_points = lighthouse_json["audits"][audit]["score"] * max_audit_points
249 | |         lost_audit_points = max_audit_points - acheived_audit_points
250 | |         points_lost[humanise_category(audit)] = round(lost_audit_points)
    | |________________________________________________________________________^ PLC0206
```

But if I correct it as recommended I get

```
helpers/lighthouse.py:246:31: PERF102 When using only the keys of a dict use the `keys()` method
    |
244 |     points_lost = {}
245 |
246 |     for audit, audit_score in perf_audits.items():
    |                               ^^^^^^^^^^^^^^^^^ PERF102
247 |         max_audit_points = perf_audits[audit]
248 |         acheived_audit_points = lighthouse_json["audits"][audit]["score"] * max_audit_points
    |
    = help: Replace `.items()` with `.keys()`
```

And if i do that:
```
helpers/lighthouse.py:246:9: SIM118 [*] Use `key in dict` instead of `key in dict.keys()`
    |
244 |     points_lost = {}
245 |
246 |     for audit in perf_audits.keys():
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^ SIM118
247 |         max_audit_points = perf_audits[audit]
248 |         acheived_audit_points = lighthouse_json["audits"][audit]["score"] * max_audit_points
    |
    = help: Remove `.keys()`
```

.... It tells me to turn it back to what I had in the beginning, so I can't win?

**Comment by dylwil3** [comment]
I think PLC0206 is suggesting that you replace the dictionary lookup with the use of the value as well. What happens if you do this in the first edit?

```python
for audit, audit_score in perf_audits.items():
        max_audit_points = audit_score
```

**Comment by njsaunders** [comment]
Aaaaah I understand - Yes thats got it - Thank you!
